### PR TITLE
Fix mac build

### PR DESCRIPTION
**Pull request by acquamarin**
1. Explicit export causes `ValueVector::setValue ` not be called correctly. Disabled explicit export on mac.
2. Fix structColumn resizing bug.

### CODE REVIEW

**Review by andyfengHKU — APPROVED** [review]

**Comment by Riolku** [comment]
Why does this happen, and why on mac only? Whats the proper solution, surely "disable explicit symbol exports" isnt the only solution.

File: CMakeLists.txt [hunk 1]
```diff
@@ -8,8 +8,10 @@ set(CMAKE_FIND_PACKAGE_RESOLVE_SYMLINKS TRUE)
 set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_CXX_STANDARD_REQUIRED True)
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+if(NOT APPLE)
```
Comment (Riolku) [inline CMakeLists.txt:11]: We should indent the body.

**Comment by acquamarin** [comment]
> Why does this happen, and why on mac only? Whats the proper solution, surely "disable explicit symbol exports" isnt the only solution.

@benjaminwinger is continuing investigating this.

File: src/processor/operator/persistent/reader/csv/driver.cpp [hunk 2]
```diff
@@ -251,7 +251,7 @@ static bool tryCastStringToStruct(const char* input, uint64_t len, ValueVector*
         skipWhitespace(++input, end);
 
         copyStringToVector(StructVector::getFieldVector(vector, fieldIdx).get(), rowToAdd,
-            std::string_view{valStart, valEnd}, csvReaderConfig);
+            std::string_view{valStart, (size_t)(valEnd - valStart)}, csvReaderConfig);
```
Comment (Riolku) [inline src/processor/operator/persistent/reader/csv/driver.cpp:254]: Do we have a test for this branch? If so, why wasn't this caught?

**Comment by Riolku** [comment]
> > Why does this happen, and why on mac only? Whats the proper solution, surely "disable explicit symbol exports" isnt the only solution.
> 
> @benjaminwinger is continuing investigating this.

Ok, let's merge this as a hotfix in the meantime.

**Pull request merged**